flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkOperatorCoordinator.java

```diff
@@ -152,9 +176,11 @@ private CoordinationResponse handleRequestImpl(
             throw new NullPointerException("No sinkAddress available.");
         }
 
-        if (socketConnection == null) {
-            socketConnection = SocketConnection.create(socketTimeout, sinkAddress);
-            LOG.info("Sink connection established");
+        synchronized (connectionLock) {
+            if (running && socketConnection == null) {
```

Comment:
otherwise we should exit eagerly instead of running into an NPE later on.

Reply:
Good point. This remark became obsolete because I removed the synchronization again.